src/main/java/org/wso2/carbon/identity/oauth2/grant/kerberos/KerberosGrant.java

```diff
@@ -383,4 +390,29 @@ private boolean isUserExistsInUserStore(String username, String tenantDomain, St
 
         return false;
     }
+
+    private ServiceProvider getServiceProvider(OAuth2AccessTokenReqDTO tokenReq) throws IdentityOAuth2Exception {
+        ServiceProvider serviceProvider;
+        try {
+            serviceProvider = OAuth2ServiceComponentHolder.getApplicationMgtService().getServiceProviderByClientId(
+                    tokenReq.getClientId(), OAuthConstants.Scope.OAUTH2, tokenReq.getTenantDomain());
```

Comment:
OAuthConstants.Scope.OAUTH2 is not the constant to use here i guess. The value we need to pass is the inbound type we are looking up.

Use [1] or [2]

[1] https://github.com/wso2/carbon-identity-framework/blob/4e311da4e8e686bd74c8abcb17763f0ef61bd5b3/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java#L300

[2] https://github.com/wso2/carbon-identity-framework/blob/4e311da4e8e686bd74c8abcb17763f0ef61bd5b3/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java#L59